DIPs/7NNN-WGB.md

```diff
@@ -0,0 +1,164 @@
+# Named Arguments
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | (number/id -- assigned by DIP Manager)                          |
+| Review Count:   | 0 (edited by DIP Manager)                                       |
+| Author:         | Walter Bright walter@digitalmars.com                            |
+| Implementation: | (links to implementation PR if any)                             |
+| Status:         | Will be set by the DIP manager (e.g. "Approved" or "Rejected")  |
+
+## Abstract
+
+Required.
+
+Short and concise description of the idea in a few lines.
+
+
+## Contents
+* [Rationale](#rationale)
+* [Prior Work](#prior-work)
+* [Description](#description)
+* [Breaking Changes and Deprecations](#breaking-changes-and-deprecations)
+* [Reference](#reference)
+* [Copyright & License](#copyright--license)
+* [Reviews](#reviews)
+
+## Rationale
+
+1. Although this supports reordering, the real reason for naming is so one can
+have
+a function with a longish list of parameters, each with a reasonable default,
+and the user need only supply the arguments that matter for his use case. This
+is much more flexible than the current method of putting all the defaults at
+the
+end of the parameter list, and defaulting one means all the rest get defaulted.
+2. For a longish list of parameters, when the user
+finds himself counting parameters to ensure they line up, then named parameters
+can be most helpful.
+3. A step towards making it possible to replace the brace struct initialization syntax with
+a function-call-like construct. I.e. the initialization of structs is unified with
+struct literals and struct constructors.
```

Comment:
Two reasons for named arguments not mentioned here:
1. Allow replacing `std.typecons.Flag`, which is a bit syntactically awkward.
2. Situations when code clarity at the caller site is of paramount importance (i.e. for some critical code), to make it unquestionably clear which values go where, even for short argument lists.

Reply:
incorporated